backend/src/shortener/shortener.controller.ts

```diff
@@ -25,12 +25,19 @@ export class ShortenerController {
   @Post()
   async shortener(@Body() body: ShortenerDTO) {
     let parsedUrl: URL;
+    let isAlreadyShortened: boolean;
+
     try {
       parsedUrl = new URL(body.originalUrl);
+
+      // Checks if the URL is not an already Reduced.to short URL.
+      isAlreadyShortened = this.shortenerService.isURLAlreadyShortend(body.originalUrl);
+      if(isAlreadyShortened) throw new Error('The URL is already shortened by reduced.to')
```

Comment:
I would write it like this, makes it more easy to understand.

```suggestion
    try {
      parsedUrl = new URL(body.originalUrl);

      // Checks if the URL is not already a reduced.
      if(this.shortenerService.isURLAlreadyShortend(body.originalUrl)) {
          throw new Error('The URL is already shortened by reduced.to')
      }